Source/ASDisplayNode+Layout.mm

```diff
@@ -127,25 +128,20 @@ - (ASTraitCollection *)asyncTraitCollection
 
 ASPrimitiveTraitCollectionDeprecatedImplementation
 
-@end
-
-#pragma mark -
 #pragma mark - ASLayoutElementAsciiArtProtocol
 
-@implementation ASDisplayNode (ASLayoutElementAsciiArtProtocol)
-
 - (NSString *)asciiArtString
 {
   return [ASLayoutSpec asciiArtStringForChildren:@[] parentName:[self asciiArtName]];
 }
 
 - (NSString *)asciiArtName
 {
-  NSString *string = NSStringFromClass([self class]);
+  NSMutableString *result = [NSMutableString stringWithCString:object_getClassName(self) encoding:NSASCIIStringEncoding];
```

Comment:
Yes I wanted to reduce the number of strings created from 3 to 1 (actually 2 since `appendFormat:` is implemented naïvely). Digging into the current implementation of `NSStringFromClass` shows that it always creates a new instance of NSString.